airflow/providers/google/cloud/example_dags/example_dataflow.py

```diff
@@ -166,7 +166,7 @@
     # [START howto_sensor_wait_for_job_status]
     wait_for_python_job_async_done = DataflowJobStatusSensor(
         task_id="wait-for-python-job-async-done",
-        job_id="{{task_instance.xcom_pull('start-python-job-async')['job_id']}}",
+        job_id="{{task_instance.xcom_pull('start-python-job-async')['id']}}",
```

Comment:
Are you sure? Please look at https://github.com/apache/airflow/blob/82de4a87b2a36e88e0b5803ed85e17d30719819a/airflow/providers/google/cloud/operators/dataflow.py#L1150
Did I miss something?

Reply:
You're right. However there is an issue for DataflowTemplatedJobStartOperator. The XCom doesn't have any job_id attribute.
To solve this, the operators/dataflow.py can be updated for DataflowTemplatedJobStartOperator (as well as flextemplateJob and sqlJob operators): return job_id instead of job.